### DISCUSSION TITLE
Should Q becomes higher as training number gets more?

### DISCUSSION THREAD

**Discussion by hokhay** [AI4Finance-Foundation/FinRL#729] [Q&A]
The graph below shows Q, actor loss and reward as training process. The model is SAC.
As far I know, the algorithm objective is to maximise Q value, so I confuse that why Q value decreases along the training. 
Also I notice that the more training episode, loss gets smaller but the result gets worsen in testing. If that's the case how many episode should I choose to train?
![image](https://user-images.githubusercontent.com/28356672/193890195-1cdebc06-d14f-4e41-9f93-27458bca282c.png)
Thanks
Jason

**Comment by alex-seto** [comment]
Hey man, this is a common problem in machine learning where if we overfit our training set, we see in testing that often performance will decrease as our algorithm will learn the training data to well. There are many ways to address this across different algorithms, I suggest googling "overfitting" and whatever algorithm you are selecting. But as I said this is a very well explored, fundamental issue in ML/RL and its something you will have to take the time to learn about thoroughly.

**Comment by zhumingpassional** [comment]
obviously not.

you should stop training to avoid overfitting.
